include/zephyr/net/prometheus/collector.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2024 Mustafa Abdullah Kus, Sparse Technology
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_PROMETHEUS_COLLECTOR_H_
+#define ZEPHYR_INCLUDE_PROMETHEUS_COLLECTOR_H_
+
+/**
+ * @file
+ *
+ * @brief Prometheus collector APIs.
+ *
+ * @defgroup prometheus Prometheus API
+ * @since 3.8
```

Comment:
There's no such thing as Zephyr 3.8 :)

```suggestion
 * @since 4.0
```